src/cpp/r/R/Options.R

```diff
@@ -41,6 +41,18 @@ if (is.null(getOption("viewer"))) {
    })
 }
 
+# default page_viewer option if not already set
+if (is.null(getOption("page_viewer"))) {
+   options(page_viewer = function(file)
+   {
+      if (!is.character(file) || (length(file) != 1))
+         stop("file must be a single element character vector.", call. = FALSE)
+
+      # show the preview window
+      invisible(.Call("rs_showPageViewer", file))
```

Comment:
I'm not sure that these reports will always require the self-contained treatment. Maybe I should just add a `self_contained` parameter which when "auto" will make a best effort to do `self_contained` but not complain otherwise.

I suggest this because my anticipated uses of this won't require self-contained so I don't want an error to occur if rmarkdown isn't available.